sdk/iot/hub/samples/src/paho_iot_hub_sas_telemetry_example.c

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef _MSC_VER
+// warning C4201: nonstandard extension used: nameless struct/union
+#pragma warning(disable : 4201)
+#endif
+#include <paho-mqtt/MQTTClient.h>
+#ifdef _MSC_VER
+#pragma warning(default : 4201)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#ifdef _WIN32
+// Required for Sleep(DWORD)
+#include <Windows.h>
+#else
+// Required for sleep(unsigned int)
+#include <unistd.h>
+#endif
+
+#include "sample_base64.h"
+
+#include <az_iot_hub_client.h>
+#include <az_result.h>
+#include <az_span.h>
+
+// TODO: #564 - Remove the use of the _az_cfh.h header in samples.
+//              Note: this is required to work-around MQTTClient.h as well as az_span init issues.
+#include <_az_cfg.h>
+
+// DO NOT MODIFY: Device ID Environment Variable Name
+#define ENV_DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// DO NOT MODIFY: IoT Hub Hostname Environment Variable Name
+#define ENV_IOT_HUB_HOSTNAME "AZ_IOT_HUB_HOSTNAME"
+
+// DO NOT MODIFY: IoT Hub SAS Key Environment Variable Name
+#define ENV_IOT_HUB_SAS_KEY "AZ_IOT_HUB_DEVICE_SAS_KEY"
+
+// DO NOT MODIFY: IoT Hub SAS Key Expiration Environment Variable Name (defaults to 2 hours)
+#define ENV_IOT_HUB_SAS_KEY_EXPIRATION "AZ_IOT_HUB_DEVICE_SAS_KEY_EXPIRATION"
+
+// DO NOT MODIFY: the path to a PEM file containing the server trusted CA
+// This is usually not needed on Linux or Mac but needs to be set on Windows.
+#define ENV_DEVICE_X509_TRUST_PEM_FILE "AZ_IOT_DEVICE_X509_TRUST_PEM_FILE"
+
+#define SAS_TOKEN_EXPIRATION_TIME_DIGITS 4
+#define TIMEOUT_MQTT_DISCONNECT_MS (10 * 1000)
+#define TELEMETRY_SEND_INTERVAL 1
+#define NUMBER_OF_MESSAGES 5
+
+static const uint8_t null_terminator = '\0';
+static char device_id[64];
+static char iot_hub_hostname[128];
+static uint32_t iot_hub_sas_key_expiration;
+static char iot_hub_sas_key[128];
+static az_span iot_hub_sas_key_span;
+static char x509_trust_pem_file[256];
+char telemetry_topic[128];
+
+static char mqtt_client_id[128];
+static char mqtt_username[128];
+static char mqtt_endpoint[128];
+static char sas_b64_decoded_key[64];
+static char sas_signature_buf[128];
+static char sas_signature_hmac_encoded_buf[128];
+static char sas_signature_encoded_buf_b64[128];
+static char sas_key_password[256];
+static az_span mqtt_url_prefix = AZ_SPAN_LITERAL_FROM_STR("ssl://");
+static az_span mqtt_url_suffix = AZ_SPAN_LITERAL_FROM_STR(":8883");
+
+static const char* telemetry_message_payloads[NUMBER_OF_MESSAGES] = {
+  "Message One", "Message Two", "Message Three", "Message Four", "Message Five",
+};
+
+static az_iot_hub_client client;
```

Comment:
there are 2 clients... hub_client?

Reply:
valid point. I can go through all of the files after this PR and update them (I think they're all named the same thing). Same with the above one on the const